### PR TITLE
Test EventListener handling of requests modified by interceptors

### PR DESCRIPTION
**Pull request by amirlivneh**
Reproduces https://github.com/square/okhttp/issues/4218.

### CODE REVIEW

File: okhttp-tests/src/test/java/okhttp3/EventListenerTest.java [hunk 1]
```diff
@@ -1125,4 +1126,28 @@ public void redirectUsingNewConnectionEventSequence() throws IOException {
         "CallEnd");
     assertEquals(expectedEvents, listener.recordedEventTypes());
   }
+
+  @Ignore("Header added by interceptor is not seen by EventListener.")
+  @Test
+  public void requestModifiedByInterceptor() throws IOException {
+    server.enqueue(new MockResponse());
+
+    client
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: You need to assign this to the client again.

Comment (amirlivneh) [reply okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: Good catch. Fixed.

**Comment by swankjesse** [comment]
This test reproduces the reported problem but I still feel that this is working as it should.  That is, I don't think there's anything to fix here.

**Comment by amirlivneh** [comment]
Yes, that's true. It's there to help the implementor of the enhancement.

**Comment by swankjesse** [comment]
Got it. I’m going to close this but keep it linked to the issue. My preference is not to check in tests for enhancements ’cause it suggests the API is decided.

**Pull request closed without merging**